### PR TITLE
Add p2p MaxMessageSize config

### LINKED ISSUE #4069: Increase p2p max message size to 12M
### What would you like to be added
Increase p2p max message size from 1M to 12M.

### Why is this needed
Per IIP24, the max theoretical message size more than 1M, the default limit of p2p. As a result, the minted block cannot be broadcasted successfully via p2p.
In v1.13.0, we may want to implement EIP4844, which may further increase the size 10 times.

### How important you think this is for IoTeX
- [x] must have
- [] should have
- [] nice to have

### Additional information
Let us know any background or context that would help us better understand the request (for example the particular use-case that prompted this request)

### PR DESCRIPTION
**Pull request by millken**
# Description
When MintNewBlock size is bigger, the p2p maxmessagesize limit will be triggered, causing the chain to get stuck.
The default in the original libp2p is `1M`, but the default is `12M` now.

Fixes #4069

## Type of change
Please delete options that are not relevant.
- [x] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] test passed on nightly
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by dustinxie — DISMISSED** [review]

**Comment by Liuhaai** [comment]
Unconfigurable from iotex core?

**Comment by dustinxie** [comment]
> Unconfigurable from iotex core?

it has a new API `WithMaxMessageSize()` which can configure from iotex-core. If not called, it now defaults to 12MB message size.

**Review by huangzhiran — APPROVED** [review]

**Comment by millken** [comment]
> Unconfigurable from iotex core?

done

File: p2p/agent.go [hunk 1]
```diff
@@ -143,6 +144,7 @@ var DefaultConfig = Config{
 	EnableRateLimit:   true,
 	PrivateNetworkPSK: "",
 	MaxPeers:          30,
+	MaxMessageSize:    p2p.DefaultConfig.MaxMessageSize,
```
Comment (envestcc) [inline p2p/agent.go:147]: I think the `12MB` should be the default parameter for the iotex-core instead of the p2p.

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**